tests/utilities/test_cli.py

```diff
@@ -689,10 +717,119 @@ def __init__(self, optim1: dict, optim2: dict, scheduler: dict):
     assert isinstance(cli.model.scheduler, torch.optim.lr_scheduler.ExponentialLR)
 
 
+@CALLBACK_REGISTRIES
+class CustomCallback(Callback):
+    pass
+
+
+def test_registries(tmpdir):
+
+    assert CALLBACK_REGISTRIES.available_objects() == [
+        "BackboneFinetuning",
+        "BaseFinetuning",
+        "BasePredictionWriter",
+        "EarlyStopping",
+        "GPUStatsMonitor",
+        "GradientAccumulationScheduler",
+        "LambdaCallback",
+        "LearningRateMonitor",
+        "ModelCheckpoint",
+        "ModelPruning",
+        "ProgressBar",
+        "ProgressBarBase",
+        "QuantizationAwareTraining",
+        "StochasticWeightAveraging",
+        "Timer",
+        "XLAStatsMonitor",
+        "CustomCallback",
+    ]
+
+    assert OPTIMIZER_REGISTRIES.available_objects() == [
+        "ASGD",
+        "Adadelta",
+        "Adagrad",
+        "Adam",
+        "AdamW",
+        "Adamax",
+        "LBFGS",
+        "RMSprop",
+        "Rprop",
+        "SGD",
+        "SparseAdam",
+    ]
+
+    assert LR_SCHEDULER_REGISTRIES.available_objects() == [
+        "CosineAnnealingLR",
+        "CosineAnnealingWarmRestarts",
+        "CyclicLR",
+        "ExponentialLR",
+        "LambdaLR",
+        "MultiStepLR",
+        "MultiplicativeLR",
+        "OneCycleLR",
+        "StepLR",
+    ]
+
+
+def test_registries_resolution(tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+
+    callbacks = [
+        dict(
+            class_path="pytorch_lightning.callbacks.Callback",
+        ),
+        dict(
+            class_path="pytorch_lightning.callbacks.Callback",
+            init_args=dict(),
+        ),
+    ]
+
+    cli_args = [
+        f"--trainer.default_root_dir={tmpdir}",
+        "--trainer.fast_dev_run=True",
+        "--trainer.progress_bar_refresh_rate=0",
+        "--optimizer Adam",
+        "--optimizer.lr 0.0001",
+        "--trainer.callbacks=LearningRateMonitor",
+        "--trainer.callbacks.logging_interval=epoch",
+        "--trainer.callbacks.log_momentum=True",
+        "--trainer.callbacks=ModelCheckpoint",
+        "--trainer.callbacks.monitor=loss",
+        f"--trainer.callbacks={json.dumps(callbacks)}",
+        "--lr_scheduler StepLR",
+        "--lr_scheduler.step_size=50",
+    ]
+
+    with mock.patch("sys.argv", ["any.py"] + cli_args):
+        cli = LightningCLI(TestModel)
+
+    assert isinstance(cli.trainer.optimizers[0], torch.optim.Adam)
+    assert len(cli.trainer.callbacks) == 4
+
+
```

Comment:
Could add a test to register new callbacks, optimizers, etc and validate it's availablity

Reply:
Done !